### PR TITLE
Fixes #1932 and converts airlock_electronics.dm to absolute paths

### LINKED ISSUE #1932: A single Airlock Electronic can complete multiple Airlocks.
As the title says. When going to create multiple Airlocks, you can use a single Airlock board to complete multiple. What should happen is you use one board and need another to finish the other Airlock. What is happening is the first one to finish eats the board but the other also gets a board inserted, effectively duplicating the Airlock board.

To reproduce, all you have to do is stand beside two or more Airlocks at the same stage, the 'Insert board' stage, and click both. You'll insert multiple of the same Electronic into the Airlocks, and can probably even remove the original for infinite boards.

Note: I discovered this issue with Windoors, not regular Airlocks. If that is somehow relevant.

### PR DESCRIPTION
**Pull request by MarinaGryphon**
* Fixed #1932 by adding an 'inuse' variable to airlock electronics and setting/checking whether it's in use when constructing doors.
* Converted airlock_electronics.dm to absolute paths from relative paths.

### CODE REVIEW

**Comment by Lohikar** [comment]
Changelog please.

**Comment by NanakoAC** [comment]
how does this bug happen exactly? Does someone just use the board on multiple doors in the four second doafter period?

If so, the simplest way to fix would be to check that the user is still holding the board before finalizing the procedure, Using locking vars can sometimes get messy

**Comment by MarinaGryphon** [comment]
Changelog added.

File: code/game/objects/structures/door_assembly.dm [hunk 1]
```diff
@@ -212,17 +212,22 @@
...
+		var/obj/item/weapon/airlock_electronics/EL = W
+		if(!EL.inuse)
+			playsound(src.loc, 'sound/items/Screwdriver.ogg', 100, 1)
+			user.visible_message("[user] installs the electronics into the airlock assembly.", "You start to install electronics into the airlock assembly.")
+			EL.inuse = 1
+			if(do_after(user, 40))
+				EL.inuse = 0
+				if(!src) return
+				user.drop_item()
+				W.loc = src
```
Comment (skull132) [inline code/game/objects/structures/door_assembly.dm]: `W.forceMove(src)`

File: code/game/objects/structures/windoor_assembly.dm [hunk 2]
```diff
@@ -167,19 +167,22 @@ obj/structure/windoor_assembly/Destroy()
...
+				var/obj/item/weapon/airlock_electronics/EL = W
+				if(!EL.inuse)
+					playsound(src.loc, 'sound/items/Screwdriver.ogg', 100, 1)
+					user.visible_message("[user] installs the electronics into the airlock assembly.", "You start to install electronics into the airlock assembly.")
+					EL.inuse = 1
+					if(do_after(user, 40))
+						EL.inuse = 0
+						if(!src) return
+						user.drop_item()
+						EL.loc = src
```
Comment (skull132) [inline code/game/objects/structures/windoor_assembly.dm]: `EL.forceMove(src)`

File: code/game/objects/structures/windoor_assembly.dm [hunk 3]
```diff
@@ -167,19 +167,22 @@ obj/structure/windoor_assembly/Destroy()
...
+						user << "<span class='notice'>You've installed the airlock electronics!</span>"
+						src.name = "Near finished Windoor Assembly"
+						src.electronics = EL
+					else
+						EL.inuse = 0
+						EL.loc = src.loc
```
Comment (skull132) [inline code/game/objects/structures/windoor_assembly.dm]: `EL.forceMove(src)`

File: code/game/objects/structures/door_assembly.dm (hunk 1 above)
Comment (skull132) [inline code/game/objects/structures/door_assembly.dm]: Also, don't use W and EL interchangeably here. You already instantiated EL. Use EL instead of W.

Comment (MarinaGryphon) [inline code/game/objects/structures/door_assembly.dm]: For the first point, that code was already there and I just converted it to absolute pathing, but I'll fix that. And for the second one, yup, easy enough to fix.

File: code/game/objects/structures/door_assembly.dm [hunk 4]
```diff
@@ -212,17 +212,20 @@
...
-			src.state = 2
-			src.name = "Near finished Airlock Assembly"
-			src.electronics = W
+		var/obj/item/weapon/airlock_electronics/EL = W
+		if(!EL.inuse)
+			playsound(src.loc, 'sound/items/Screwdriver.ogg', 100, 1)
+			user.visible_message("[user] installs the electronics into the airlock assembly.", "You start to install electronics into the airlock assembly.")
+			EL.inuse = 1
+			if(do_after(user, 40))
+				if(!src) return
```
Comment (skull132) [inline code/game/objects/structures/door_assembly.dm]: This and line 177 in windoor_assembly.dm aren't resetting EL.inuse to 0. Would render them unusable.

Comment (MarinaGryphon) [inline code/game/objects/structures/door_assembly.dm]: I don't see an issue with `windoor_assembly.dm`. It's set to zero before the return either way, and if the `do_after` fails, it's still set to zero. I also am going to fix it to do the same thing with `door_assembly.dm`.

File: code/game/objects/structures/door_assembly.dm [hunk 5]
```diff
@@ -212,17 +212,23 @@
...
+				user.drop_item()
+				EL.forceMove(src)
+				user << "<span class='notice'>You installed the airlock electronics!</span>"
+				src.state = 2
+				src.name = "Near finished Airlock Assembly"
+				src.electronics = EL
+			else
+				EL.inuse = 0
+				EL.forceMove(src)
```
Comment (skull132) [inline code/game/objects/structures/door_assembly.dm]: Why are you using forceMove() here? The else clause is a failed operation, surely the electronics should still be in the hands of the person installing them.

While yes, old code does `W.loc = src.loc` (note, src != src.loc), you might as well just keep them in the hands of the operator. That is to say. Not do the forceMove at all. So either fix this line to be `EL.forceMove(src.loc)` or just nuke it completely.

File: code/game/objects/structures/windoor_assembly.dm [hunk 6]
```diff
@@ -167,19 +167,22 @@ obj/structure/windoor_assembly/Destroy()
...
+						EL.inuse = 0
+						if(!src) return
+						user.drop_item()
+						EL.forceMove(src)
+						user << "<span class='notice'>You've installed the airlock electronics!</span>"
+						src.name = "Near finished Windoor Assembly"
+						src.electronics = EL
+					else
+						EL.inuse = 0
+						EL.forceMove(src)
```
Comment (skull132) [inline code/game/objects/structures/windoor_assembly.dm]: Same deal here, as above.

**Pull request merged**